src/ReactiveUI.XamForms/ShellViewModel.cs

```
@@ -0,0 +1,89 @@
+            typeof(string),
+            typeof(ShellViewModel<TViewModel>),
+            null);
+
+        /// <summary>
+        /// The view model property.
+        /// </summary>
+        public static readonly BindableProperty ViewModelProperty = BindableProperty.Create(
+            nameof(ViewModel),
+            typeof(TViewModel),
+            typeof(ShellViewModel<TViewModel>),
+            default(TViewModel),
+            BindingMode.Default,
+            propertyChanged: ViewModelChanged);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ShellViewModel{TViewModel}"/> class.
+        /// </summary>
+        public ShellViewModel()
+        {
+            ViewLocator = Locator.Current.GetService<IViewLocator>();
+            ViewModel = new TViewModel();
+        }
+
+        /// <summary>
+        /// Gets or sets the view model.
+        /// </summary>
+        /// <value>
+        /// The view model.
+        /// </value>
+        public TViewModel ViewModel
+        {
+            get => (TViewModel)GetValue(ViewModelProperty);
+            set => SetValue(ViewModelProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the contract.
+        /// </summary>
+        /// <value>
+        /// The contract.
+        /// </value>
+        public string Contract
+        {
+            get => (string)GetValue(ContractProperty);
+            set => SetValue(ContractProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the view locator.
+        /// </summary>
+        /// <value>
+        /// The view locator.
+        /// </value>
+        public IViewLocator? ViewLocator { get; set; }
+
+        private static void ViewModelChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            if (bindable is ShellViewModel<TViewModel> svm)
+            {
+                var view = svm.ViewLocator?.ResolveView(newValue as TViewModel, svm.Contract);
```

Seeing as how this class explicitly depends on the `ViewLocator` from the `Locator` I think we should throw an exception if it's null.

---

now throws new NullReferenceException(nameof(Locator.Current));